velox/dwio/common/compression/Compression.h

```diff
@@ -60,37 +67,62 @@ class Decompressor {
   const std::string streamDebugInfo_;
 };
 
+struct CompressionOptions {
+  /// Format specific compression/decompression options
+  union Format {
+    struct {
+      /// Window bits determines the history buffer size and whether
+      /// header/trailer is added to the compression block.
+      int windowBits;
+      /// Compression level determines the compression ratio. Zlib supports
+      /// values ranging from 0 (no compression) to 9 (max compression)
+      int32_t compressionLevel;
+    } zlib;
+
+    struct {
+      int32_t compressionLevel;
+    } zstd;
+  } format;
+
+  uint32_t compressionThreshold;
+};
+
 /**
  * Create a decompressor for the given compression kind.
  * @param kind the compression type to implement
  * @param input the input stream that is the underlying source
  * @param bufferSize the maximum size of the buffer
  * @param pool the memory pool
+ * @param options compression options to use
+ * @param useRawDecompression specify whether to perform raw decompression
```

Comment:
This parameter is only for GZIp and ZLIB, can you mention it in the comment?

Reply:
It is currently for Gzip and Zlib, but can be used for any format. Do we want to change the comment?

Reply:
Also compressionThreshold is not specific to these formats.